rainier-core/src/test/scala/com/stripe/rainier/compute/RealTest.scala

```diff
@@ -7,24 +7,30 @@ class RealTest extends FunSuite {
   def run(description: String)(fn: Real => Real): Unit = {
     test(description) {
       val x = new Variable
+      val x2 = new Variable
       val result = fn(x)
+      val result2 = fn(x2)
       val c = Compiler.default.compile(List(x), result)
       List(1.0, 0.0, -1.0, 2.0, -2.0, 0.5, -0.5).foreach { n =>
         val constant = fn(Constant(n))
-        assert(constant.isInstanceOf[Constant], s"[n=$n]")
-        val eval = new Evaluator(Map(x -> n))
+        assert(constant.isInstanceOf[Constant], s"[c, n=$n]")
+        val eval = new Evaluator(Map(x -> n, x2 -> n))
         val withVar = eval.toDouble(result)
         assertWithinEpsilon(constant.asInstanceOf[Constant].value,
                             withVar,
-                            s"[n=$n]")
+                            s"[c/ev, n=$n]")
         val compiled = c(Array(n))
-        assertWithinEpsilon(withVar, compiled, s"[ir, n=$n]")
+        assertWithinEpsilon(withVar, compiled, s"[ev/ir, n=$n]")
+        val withVar2 = eval.toDouble(result2)
+        assertWithinEpsilon(withVar, withVar2, s"[ev/ev, n=$n]")
```

Comment:
OOC, what failure mode is this test getting at?

Reply:
This is actually sorta aspirational in that I don't yet have any sufficiently simple test cases that I can get this to fail with. But the idea was to test numerical instability in results from random reordering of terms when building the DAG.